### ISSUE TITLE
[BUG] Individual resistive heaters are classified as central (district heating) in the OPF

### ISSUE THREAD

**Issue by MoritzSchloesser** [openego/eDisGo#759]
**Type:** bug (misclassification → wrong model parameters / possible crash)
**Affects:** `edisgo/io/powermodels_io.py::_build_heat_storage` (~lines 1333, 1361, 1402)
**Inconsistent with:** `edisgo/network/overlying_grid.py::distribute_overlying_grid_requirements` (~lines 357, 380)

## Background

Heat pumps carry a `sector` in `topology.loads_df` with four permitted values
(documented in `network/topology.py` ~line 274):

| `sector` | device |
|---|---|
| `individual_heating` | heat pump serving one building |
| `individual_heating_resistive_heater` | resistive heater serving one building |
| `district_heating` | heat pump feeding a district heating network |
| `district_heating_resistive_heater` | resistive heater feeding a district heating network |

"Decentral" is the two `individual_*` values; "central" the two `district_*` values.

## Problem

Two code paths perform this split and disagree.

`distribute_overlying_grid_requirements` uses explicit lists and is correct:

```python
# overlying_grid.py ~357 / ~380
loads_df.sector.isin(["district_heating", "district_heating_resistive_heater"])
loads_df.sector.isin(["individual_heating", "individual_heating_resistive_heater"])
```

`_build_heat_storage` in the OPF instead defines central by negation:

```python
# powermodels_io.py ~1361
central_hps = np.intersect1d(
    loads_df.loc[(loads_df.type == "heat_pump")
                 & (loads_df.sector != "individual_heating")].index,   # <-- catches 3 values
    flexible_hps,
)
```

`sector != "individual_heating"` matches **three** of the four values, so
`individual_heating_resistive_heater` — a building-level device — is grouped with
district heating units. The decentral selection immediately above
(`sector == "individual_heating"`, ~line 1333) excludes it symmetrically, so it is not
merely double-counted: it is classified as central only.

## Consequences

**1. Lookup by a district heating ID the unit does not have.** Central heat pumps are
matched to their district heating area's thermal storage SoC (~line 1365):

```python
data = edisgo_obj.overlying_grid.thermal_storage_units_central_soc[
    edisgo_obj.topology.loads_df.loc[central_hps]
    .district_heating_id.astype(int).astype(str)
].values
```

`district_heating_id` is NaN for individual units, so `.astype(int)` raises
(`ValueError`/`IntCastingNaNError`) when `thermal_storage_units_central_soc` is
populated and an individual resistive heater is in `flexible_hps`. Where that
overlying-grid series is empty the branch is skipped and the error is masked — so the
failure is intermittent, depending on the input data.

**2. Wrong standing-loss factor.** The same negation sets the thermal storage loss
(~line 1402):

```python
if loads_df.loc[heat_storage_df.index[stor_i]].sector != "individual_heating":
    p_loss = 0          # district heating storage
else:
    p_loss = 0.04       # 4% of SoC per day
```

An individual resistive heater's storage is therefore modelled as lossless instead of
losing 4%/day.

**3. The two code paths are not comparable.** In any grid containing individual
resistive heaters, results from the rule-based distribution and from the OPF split the
central/decentral overlying-grid requirements differently, so they cannot be validated
against each other.

## Suggested fix

Replace both negations with the explicit list already used in `overlying_grid.py`:

```python
CENTRAL_SECTORS = ["district_heating", "district_heating_resistive_heater"]

central_hps = np.intersect1d(
    loads_df.loc[(loads_df.type == "heat_pump")
                 & loads_df.sector.isin(CENTRAL_SECTORS)].index,
    flexible_hps,
)
```

and correspondingly for the `p_loss` branch. Defining the two sector lists once as
module-level constants shared by both files would prevent the pair from drifting again.

Consider also raising on any heat pump whose sector is in neither list, so an
unexpected value surfaces instead of silently landing in the central group.

## Detection

```python
loads = edisgo.topology.loads_df
affected = loads.index[(loads.type == "heat_pump")
                       & (loads.sector == "individual_heating_resistive_heater")]
print(f"{len(affected)} individual resistive heaters affected")
```

Grids without individual resistive heaters are unaffected, since the negation and the
explicit list agree on the other three values.
